hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/Util.java

```diff
@@ -106,6 +116,47 @@ public static String getMsgType(String hlMsg) {
 		}
 		return msgType;
 	}
+	
+	/**
+	 * returns the message id based on the HL7 message.
+	 * @param hlMsg
+	 * @return
+	 */
+	public static String getMsgId(String hlMsg) {
+
+		String msgId = "";
+
+		if (hlMsg == null || hlMsg.isEmpty()) {
+			return msgId;
+		}
+
+		String[] hl7MessageAtt = hlMsg.split(DOUBLE_BACKSLASH + HL7_DELIMITER);
+		if (hl7MessageAtt.length > 9) {
+			msgId = hl7MessageAtt[9];
+		}
+	
+		return msgId;
+	}
+	
+	/*
+	 * The FacilityId is the legacy way to track connected clients and is now set as
+	 * the ClientId of the client application In the access token this is the 'azp'
+	 * field
+	 */
+	public static String getSendingFacility(String auth) {
+		String clientId = "";
+		if (!StringUtils.isEmpty(auth)) {
```

Comment:
I prefer to return early in situations like these as it avoids having your logic buried in an if statement. I.e. take the approach you used in getMsgId where you return right away if the value is empty/null.

Reply:
done